core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/core/jdbc/dataaccess/JDBCTransactionManager.java

```diff
@@ -94,6 +94,14 @@ public void beginTransaction() throws RegistryException {
         } catch (SQLException e) {
             String msg = "Failed to start new registry transaction.";
             log.error(msg, e);
+            if (conn != null) {
+                try {
+                    conn.close();
+                } catch (SQLException e1) {
+                    String msgFailed = "Failed to close connection.";
+                    log.error(msgFailed, e);
+                }
+            }
```

Comment:
Thanks for your contribution! As it looks like still there is a possibility for the connection not to be closed if temp becomes null. 

I think it might be better to move this block to a "finally" clause. Thanks!